abis/linux/msg.h

```diff
@@ -0,0 +1,33 @@
+#ifndef _ABIBITS_MSG_H
+#define _ABIBITS_MSG_H
+
+#include <sys/ipc.h>
+#include <bits/ansi/time_t.h>
+#include <abi-bits/pid_t.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef unsigned long msglen_t;
+typedef unsigned long msgqnum_t;
```

Comment:
should we make these fixed-width? how do they differ across the arches we have?

Reply:
okay, I looked around the logic here a bit, I think these are always `unsigned long int`, but they depend on `__WORDSIZE`, and I'm not sure how we handle that case

Reply:
We can copy the logic that depends on __WORDSIZE in a static assert.

Reply:
This comment is still not addressed, maybe @ArsenArsen can link to the logic in question so it's easier to copy over?

Reply:
Looking through glibc behavior, these are always defined to be `unsigned long int`, except for i386, where they are `uint64_t`.